### DISCUSSION TITLE
RTSP guidence needed

### DISCUSSION THREAD

**Discussion by redryder71** [gtxaspec/wz_mini_hacks#734] [Q&A]
Cam V3 with fw 4.36.9.139, wz_mini_hacks, and tinyCam 17.2.1 - cam has fixed address.
Trying to make RTSP work on tinyCam to cut out the Wyze Cloud and run local only.

current config rtsp section of config, there is a pw set
#####VIDEO STREAM#####
RTSP_LOGIN="admin"
RTSP_PASSWORD="****"
RTSP_PORT="8554"

RTSP_HI_RES_ENABLED="false"
RTSP_HI_RES_ENABLE_AUDIO="false"

RTSP_LOW_RES_ENABLED="false"
RTSP_LOW_RES_ENABLE_AUDIO="false"

RTSP_AUTH_DISABLE="false"

VIDEO_0_MAX_BITRATE=""
VIDEO_0_TARGET_BITRATE=""
VIDEO_0_ENC_PARAMETER=""
VIDEO_0_FPS=""

VIDEO_1_MAX_BITRATE=""
VIDEO_1_TARGET_BITRATE=""
VIDEO_1_ENC_PARAMETER=""
VIDEO_1_FPS=""

GO2RTC_SERVER_ENABLED="true"
GO2RTC_HI_RES_ENABLED="true"
GO2RTC_HI_RES_ENABLE_AUDIO="true"
GO2RTC_LOW_RES_ENABLED="true"
GO2RTC_LOW_RES_ENABLE_AUDIO="true"

The IP address, port, account and pw match whats in the config 
tinyCam give the error "Video failed (Timeout, check hostname, and port number)"

Cam does work on the Wyze cloud protocol 
If I try to connect from browser
"Firefox can’t establish a connection to the server at 192.168.0.163:1984"

What should I try next?

**Comment by jmbulan** [comment]
Are you able to access the go2rtc server at all?  It provides different stream links for various codecs

**Comment by redryder71** [comment]
If the correct path is http://192.168.0.163:1984/  or https://192.168.0.163:1984/  - still get  "can’t establish a connection " error.

**Reply by jmbulan** [reply]
Are you able to ssh into the camera? What build of wzmini are you on?

**Comment by redryder71** [comment]
Did not try SSH  - wanted to keep the variables as low as possible.
Believe the latest version of wz_mini_hacks as I pulled the files on Saturday  2-24-24

**Comment by redryder71** [comment]
Started fresh with a  new formatted card, pulled down the software this morning, and copied over the files as per the install directions. The  go2rtc server is found and I can view stream in web browser. tinyCam Monitor still can't get the RTSP. Also tried Agent DVR but no luck there either.  

tinyCam a "Camera Status" button that is the place where the "Video failed (Timeout, check hostname, and port number)". It also has an advanced info line where it reports the url as rtsp://192.168.0.163:8554/live. Could tinyCam be expecting an exact match stream from the cam? 

Is there a more suitable camera viewer then tinyCam that runs on Android devices? 
Thanks

**Reply by Bob-AOL** [reply]
I use tinyCam with no issues, though with an older version of wz_mini with the older RTSP server. Regardless, it wants to see the proper MRL (Media Resource Locator) address and associated user/pass credentials. The 'xxxx/live' format is the Wyze Beta RTSP firmware format. You basically just need to specify the IP address, port 80 for http, 1984 for rtsp, and the correct user/pass credentials and tinyCam will do the rest. If selecting Wyze for brand and model in tinyCam, try generic/onvif instead. It's probably defaulting to  the Wyze format for the MRL.

Another similar Android player/viewer is Onvier. The demo is very frustrating to use, but it only costs 8 or 9 bucks to buy. I personally prefer tinyCam over any of the others and use it in several locations with 22 cameras.

**Reply by clee604** [reply]
for some of my older android devices, I use the webrtc stream from the go2rtc server because it seems chrome has better hw acceleration support than tinycam. I just set up bookmarks with a list of my cameras and open as needed.

**Comment by gtxaspec** [comment]
you should be adding the camera as a "custom" model in tinycam, then entering the URL's manually.

**Reply by Bob-AOL** [reply]
I believe that would actually be 'generic'. At least it is in my version. (No 'custom' listed.)

Basically my badly attempted suggestion above....

Your clarification should be quite helpful in clearing that up. LOL
